pythonforandroid/recipes/fortran/__init__.py

```diff
@@ -0,0 +1,188 @@
+import os
+import subprocess
+import shutil
+import sh
+from pathlib import Path
+from os.path import join
+from pythonforandroid.recipe import Recipe
+from pythonforandroid.recommendations import read_ndk_version
+from pythonforandroid.logger import info, shprint, info_main
+from pythonforandroid.util import ensure_dir
+import hashlib
+
+FLANG_FILES = {
+    "package-flang-aarch64.tar.bz2": "775f362c758abe8d3173edc7be9ced3730ff14c64d44743017c3af7ceb0a6610",
+    "package-flang-host.tar.bz2": "04fe24d67ee7eb5a4223299c610013585e75c56467e4b185ed929a3d17e3d077",
+    "package-flang-x86_64.tar.bz2": "2061a0e3179f4afa55516ce3858582d25ea7b108ff762d9fb4ec8a03b49b36d2",
+    "package-install.tar.bz2": "d37dc6a58b495807f015c7fec08a57ff95d52ad0d0553cbf573b0215d8a1707c",
+}
+
+
+class GFortranRecipe(Recipe):
+    # flang support in NDK by @licy183 (on github)
+    name = "fortran"
+    toolchain_ver = 0
+    url = "https://github.com/licy183/ndk-toolchain-clang-with-flang/releases/download/"
+
+    def match_sha256(self, file_path, expected_hash):
+        sha256 = hashlib.sha256()
+        with open(file_path, "rb") as f:
+            for chunk in iter(lambda: f.read(8192), b""):
+                sha256.update(chunk)
+        file_hash = sha256.hexdigest()
+        return file_hash == expected_hash
```

Comment:
I'm not sure how to feel about downloading a binary from a non-official source.
It's indeed very convenient to have the option, but maybe we can have an extra runtime warning or yes/no prompt for the user before moving forward with the recipe download/install, what do you think?
I'm not sure what's the best approach, we don't want the implementation to be cumbersome of course, but we still want to protect our users.
Or maybe none of this, but enforce hash verification for this recipe, with a code comment to future developer explaining that the recipe should always come with the hash check (assuming due diligence was done by the recipe maintainer upfront for spot checking a version before offering it).
I'm open to suggestion and debate the options

Reply:
> non-official source

Changed to termux one in latest commit.

> enforce hash verification for this recipe,

Already enforced.

Btw it's just a normal download url as other recipes.